### PR TITLE
ui: CopyButton amends

### PR DESCRIPTION
**Pull request by johncowen**
During https://github.com/hashicorp/consul/pull/10285 we noticed that our CopyButton needed a few tweaks, luckily we have also decided to update how this works slightly.

This PR refactors the copy-to-clipboard functionality out of our CopyButton component and into a 'non-visual' modifier where it probably belongs.

- [with-copyable](https://consul-ui-staging-defwazcw9-hashicorp.vercel.app/ui/docs/modifiers/with-copyable)
- [CopyButton](https://consul-ui-staging-defwazcw9-hashicorp.vercel.app/ui/docs/components/copy-button)

Benefits:

- We don't have to reuse our CopyButton component if we want a method of copying text that doesn't look the same as our CopyButton (design vs behaviour)
- Our CopyButton component (as it already uses our StateChart component to deal with behaviour/state) is now almost a lighter-weight templateOnly component, and we should be able to make it 100% template only pretty soon.

We also:

- Moved our styles for this from our 'all-the-buttons' CSS file into the copy-button component by itself (resisted tweaking this for the moment as we are looking to change the design/look and feel of it)
- Added all the copy to our translations
- Added documentation for both `{{with-copyable}}` and `<CopyButton />`, including some info on our acceptance test steps that we use specifically for testing copying into our documentation (more related things to come here 🤞 )

Redesign will be in a follow up PR.

### CODE REVIEW

File: ui/packages/consul-ui/app/modifiers/with-copyable.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# with-copyable
```
Comment (kaxcode) [inline ui/packages/consul-ui/app/modifiers/with-copyable.mdx:1]: this great! 📝

**Review by kaxcode — APPROVED** [review]
ok, I see it's mostly moving things around and adding more documentation. The `with-copyable` modifier will come in handy if we create another `copy` component. LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/404188.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit a6996b6ea5dcbc72fc8c3a277920118f9f6b41fc onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/404188)

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/404436.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit a6996b6ea5dcbc72fc8c3a277920118f9f6b41fc onto `release/1.10.x` succeeded!

**Pull request merged**